Actions/Github-Helper.psm1

```diff
@@ -1099,3 +1139,90 @@ function DownloadArtifact {
+ .Parameter privateKey
+  The GitHub App Private Key
+ .PARAMETER api_url
+  The GitHub API URL
+ .PARAMETER repository
+  The Current GitHub repository
+ .PARAMETER repositories
+  The repositories to request access to
+ .PARAMETER permissions
+  The permissions to request for the Access Token
+#>
+function GetGitHubAppAuthToken {
+    Param(
+        [string] $gitHubAppClientId,
+        [string] $privateKey,
+        [string] $api_url = $ENV:GITHUB_API_URL,
+        [string] $repository,
+        [hashtable] $permissions = @{},
+        [string[]] $repositories = @()
+    )
+
+    Write-Host "Using GitHub App with ClientId $gitHubAppClientId for authentication"
+    $jwt = GenerateJwtForTokenRequest -gitHubAppClientId $gitHubAppClientId -privateKey $privateKey
+    $headers = @{
+        "Accept" = "application/vnd.github+json"
+        "Authorization" = "Bearer $jwt"
+        "X-GitHub-Api-Version" = "2022-11-28"
+    }
+    Write-Host "Get App Info $api_url/repos/$repository/installation"
+    $appinfo = Invoke-RestMethod -Method GET -UseBasicParsing -Headers $headers -Uri "$api_url/repos/$repository/installation"
+    $body = @{}
+    # If repositories are provided, limit the requested repositories to those
+    if ($repositories) {
+        $body += @{ "repositories" = @($repositories | ForEach-Object { $_.SubString($_.LastIndexOf('/')+1) } ) }
+    }
+    # If permissions are provided, limit the requested permissions to those
+    if ($permissions) {
+        $body += @{ "permissions" = $permissions }
+    }
+    Write-Host "Get Token Response $($appInfo.access_tokens_url) with $($body | ConvertTo-Json -Compress)"
+    $tokenResponse = Invoke-RestMethod -Method POST -UseBasicParsing -Headers $headers -Body ($body | ConvertTo-Json -Compress) -Uri $appInfo.access_tokens_url
+    Write-Host "return token"
+    return $tokenResponse.token, $tokenResponse.expires_in
+}
+
+<#
+ .SYNOPSIS
+  Generate JWT for token request
+  As documented here: https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+ .PARAMETER gitHubAppClientId
+  The GitHub App Client ID
+ .Parameter privateKey
+  The GitHub App Private Key
+#>
+function GenerateJwtForTokenRequest {
+    Param(
+        [string] $gitHubAppClientId,
+        [string] $privateKey
+    )
+
+    $header = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes((ConvertTo-Json -InputObject @{
+        alg = "RS256"
+        typ = "JWT"
+    }))).TrimEnd('=').Replace('+', '-').Replace('/', '_');
+
+    $payload = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes((ConvertTo-Json -InputObject @{
+        iat = [System.DateTimeOffset]::UtcNow.AddSeconds(-10).ToUnixTimeSeconds()
+        exp = [System.DateTimeOffset]::UtcNow.AddMinutes(10).ToUnixTimeSeconds()
+        iss = $gitHubAppClientId
+    }))).TrimEnd('=').Replace('+', '-').Replace('/', '_');
+    $signature = pwsh -command {
```

Comment:
Consider using [Start-Job](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/start-job) as I remember there could be issues if running different processes like this. E.g. any output from the script will be added to the return value. 
Even if no output is added, there may be some output coming from the PowerShell profile ($PROFILE).

Reply:
Just FYI, `Start-Job` has `-PSVersion` parameter.

Your solution works as well.

Reply:
Correct, but that only supports older versions:
![{0DB61B30-49AF-4F44-98E1-0384FB74E493}](https://github.com/user-attachments/assets/bf8f5736-bb46-427b-8de5-15738b6301fa)